srcpkgs/mold/patches/00-grep.patch

```diff
@@ -0,0 +1,279 @@
+--- a/test/build-id.sh
++++ b/test/build-id.sh
+@@ -12,7 +12,7 @@ readelf -n $t/exe | grep -qv 'GNU.*0x00000010.*NT_GNU_BUILD_ID'
+ 
+ clang -o $t/exe $t/a.c -fuse-ld=`pwd`/../mold -Wl,-build-id=uuid
+ readelf -nW $t/exe |
+-  grep -Pq 'GNU.*0x00000010.*NT_GNU_BUILD_ID.*Build ID: ............4...[89abcdef]'
++  rg -Pq 'GNU.*0x00000010.*NT_GNU_BUILD_ID.*Build ID: ............4...[89abcdef]'
+ 
+ clang -o $t/exe $t/a.c -fuse-ld=`pwd`/../mold -Wl,-build-id=md5
+ readelf -n $t/exe | grep -q 'GNU.*0x00000010.*NT_GNU_BUILD_ID'
+--- a/test/copyrel-relro.sh
++++ b/test/copyrel-relro.sh
+@@ -22,7 +22,7 @@ EOF
+ clang -fuse-ld=`pwd`/../mold $t/a.o $t/b.so -o $t/exe
+ readelf -a $t/exe > $t/log
+ 
+-grep -Pqz '(?s)\[(\d+)\] .dynbss.rel.ro .* \1 readonly' $t/log
+-grep -Pqz '(?s)\[(\d+)\] .dynbss .* \1 readwrite' $t/log
++rg -PUq '(?s)\[(\d+)\] .dynbss.rel.ro .* \1 readonly' $t/log
++rg -PUq '(?s)\[(\d+)\] .dynbss .* \1 readwrite' $t/log
+ 
```

Comment:
You can avoid the whole patch by `ln -s /usr/bin/rg fake-bin/grep` and put it into `$PATH`

Reply:
@sgn clever solution, I appreciate it

Reply:
> You can avoid the whole patch by `ln -s /usr/bin/rg fake-bin/grep` and put it into `$PATH`

@sgn I think the patch is needed anyway because sometimes `grep` & `rg` have slightly different options: `-U` in `rg` vs  `-z` in `grep` for multiline mode. Though I can make the patch a lot smaller if I employ your approach. Should I do that?